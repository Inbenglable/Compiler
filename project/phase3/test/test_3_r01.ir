FUNCTION mod :
PARAM var_x
PARAM var_n
tmp_1 := var_x
tmp_5 := var_x
tmp_6 := var_n
tmp_3 := var_x / var_n
tmp_4 := var_n
tmp_2 := tmp_3 * var_n
tmp_0 := var_x - tmp_2
RETURN tmp_0
FUNCTION isPalindrome :
PARAM var_num
var_reversed := #0
var_original := var_num
var_digit := #0
LABEL label_0 :
tmp_7 := var_num
tmp_8 := #0
IF tmp_7 != tmp_8 GOTO label_1
GOTO label_2
LABEL label_1 :
tmp_10 := var_num
tmp_11 := #10
ARG tmp_11
ARG tmp_10
tmp_9 := CALL mod
var_digit := tmp_9
tmp_15 := var_reversed
tmp_13 := var_reversed * #10
tmp_14 := var_digit
tmp_12 := tmp_13 + var_digit
var_reversed := tmp_12
tmp_18 := var_num
tmp_17 := var_num / #10
var_num := tmp_17
GOTO label_0
LABEL label_2 :
tmp_20 := var_reversed
tmp_21 := var_original
IF tmp_20 == tmp_21 GOTO label_3
GOTO label_4
LABEL label_3 :
tmp_22 := #1
RETURN tmp_22
GOTO label_5
LABEL label_4 :
tmp_23 := #0
RETURN tmp_23
LABEL label_5 :
FUNCTION main :
READ tmp_24
var_number := tmp_24
tmp_27 := var_number
ARG tmp_27
tmp_25 := CALL isPalindrome
tmp_26 := #1
IF tmp_25 == tmp_26 GOTO label_6
GOTO label_7
LABEL label_6 :
tmp_28 := #1
WRITE tmp_28
GOTO label_8
LABEL label_7 :
tmp_29 := #-1
WRITE tmp_29
LABEL label_8 :
tmp_31 := #0
RETURN tmp_31
