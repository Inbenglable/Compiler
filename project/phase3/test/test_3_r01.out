Error type 2 at line 24: "read" is invoked without a definition
Error type 2 at line 26: "write" is invoked without a definition
Error type 2 at line 28: "write" is invoked without a definition
